broker-daemon/block-order-worker/index.js

```diff
@@ -340,21 +342,40 @@ class BlockOrderWorker extends EventEmitter {
 
     this.logger.info('Creating order for BlockOrder', { blockOrderId: blockOrder.id })
 
-    const order = await OrderStateMachine.create(
+    const osm = await OrderStateMachine.create(
       {
         relayer,
         engines,
         logger,
-        store,
-        onRejection: (err) => {
-          this.failBlockOrder(blockOrder.id, err)
-        }
+        store
       },
       blockOrder.id,
       { side, baseSymbol, counterSymbol, baseAmount, counterAmount }
     )
 
-    this.logger.info('Created order for BlockOrder', { blockOrderId: blockOrder.id, orderId: order.orderId })
+    // We are hooking into the complete lifecycle event of an order state machine to trigger
+    // the completion of a blockorder
+    osm.once('complete', async () => {
+      this.checkBlockOrderCompletion(blockOrder.id)
+        .catch(e => {
+          this.logger.error(`BlockOrder failed to be completed from order`, { id: blockOrder.id, error: e.stack })
+        })
+        .then(() => osm.removeAllListeners())
+    })
+
+    // We are hooking into the reject lifecycle event of an order state machine to trigger
+    // the failure of a blockorder
+    osm.once('reject', async () => {
+      this.failBlockOrder(blockOrder.id, osm.order.error)
+        .catch(e => {
+          this.logger.error(`BlockOrder failed on setting a failed status from order`, { id: blockOrder.id, error: e.stack })
+        })
+        .then(() => {
+          return osm.removeAllListeners()
```

Comment:
[nit] weird that you did multiple lines on this one but single line for `complete`